### PR TITLE
Begin simplification of tagging keeps

### PR DESCRIPTION
**Pull request by andrewconner**
Rushing this to merge without my new stuff in order to avoid conflicts. Just pulled master, so hopefully everyone stays happy.

• Removes unused tagging methods
• Makes `getOrCreateTag` take a db session, so batch processing becomes possible (trivia: we get tons of deadlocks here; it's because we had session-in-session)
• Creating/modifying tags happens in a much smaller surface area. Immediately, lets me change tags to become note-centric (we deal with notes, not notes and a separate set of tags). Future, we can change how tags work, and split them out if we need to more easily.

@ryanpbrewster

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 1]
```diff
@@ -530,122 +494,26 @@ class KeepCommanderImpl @Inject() (
...
-    persistHashtagsAndSaveKeep(userId, keep, selectedTagNames)(session, context)
+  // Given set of tags and keep, update keep note to reflect tag set (create tags, remove tags, insert into note, remove from note)
+  def persistHashtagsForKeepAndSaveKeep(userId: Id[User], keep: Keep, allTagsKeepShouldHave: Seq[String])(implicit session: RWSession, context: HeimdalContext) = {
+    persistHashtagsAndSaveKeep(userId, keep, allTagsKeepShouldHave)(session, context)
     searchClient.updateKeepIndex()
   }
 
   // Changes a keep's notes based on the hashtags to persist!
-  private def persistHashtagsAndSaveKeep(userId: Id[User], keep: Keep, selectedTagNames: Seq[String])(implicit session: RWSession, context: HeimdalContext) = {
+  private def persistHashtagsAndSaveKeep(userId: Id[User], keep: Keep, allTagsKeepShouldHave: Seq[String])(implicit session: RWSession, context: HeimdalContext) = {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala:514]: Hopefully renaming this variable makes it clear how this was meant to be used.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryAnalytics.scala [hunk 2]
```diff
@@ -213,65 +213,6 @@ class LibraryAnalytics @Inject() (
     }
   }
 
-  def renamedTag(oldTag: Collection, newTag: Collection, context: HeimdalContext): Unit = {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/LibraryAnalytics.scala:216]: Removing these authorized by Jen. _Floods_ threadpool during imports (each spawns a future in the general purpose threadpool, for EVERY tag on every keep).

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 3]
```diff
@@ -72,17 +72,14 @@ trait KeepCommander {
   def persistKeep(k: Keep, users: Set[Id[User]], libraries: Set[Id[Library]])(implicit session: RWSession): Keep
 
   // Updating / managing
-  def updateKeepInLibrary(keepId: ExternalId[Keep], libId: Id[Library], userId: Id[User], title: Option[String])(implicit context: HeimdalContext): Either[(Int, String), Keep]
+  def updateKeepTitle(keepId: ExternalId[Keep], libId: Id[Library], userId: Id[User], title: Option[String])(implicit context: HeimdalContext): Either[(Int, String), Keep]
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala:75]: You're my favorite

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 4]
```diff
@@ -448,8 +413,7 @@ class KeepCommanderImpl @Inject() (
     searchClient.updateKeepIndex()
   }
 
-  // TODO(ryan): Murder whoever decided "updateKeep" was descriptive
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala:451]: :grin:

**Comment by ryanpbrewster** [comment]
lgtm, I love how we're cleaning up this stuff

**Comment by ryanpbrewster** [comment]
Jesus, 800+ lines removed, you lucky dog.

**Pull request merged**